Snippets/MsLogging/MsLogging_1/ProgramService.cs

```diff
@@ -37,8 +38,16 @@ protected override void OnStart(string[] args)
 
     async Task AsyncOnStart()
     {
-        loggerFactory = new LoggerFactory();
-        loggerFactory.AddConsole();
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddLogging(loggingBuilder =>
+        {
+            loggingBuilder.AddFilter(level => level >= Microsoft.Extensions.Logging.LogLevel.Information);
+            loggingBuilder.AddConsole();
+        });
+
+        var serviceProvider = serviceCollection.BuildServiceProvider();
+
+        loggerFactory = serviceProvider.GetService<Microsoft.Extensions.Logging.ILoggerFactory>();
```

Comment:
this looks correct. but wow that is so much more verbose. can u point me to where u got this from? is there some ms logging upgrade guide?

Reply:
It was totally fumbling around in the dark just trying to make something work. The only thing the obsolete message says is to use ILoggingBuilder instead of ILoggerFactory. In an aspnetcore app it’s a lot easier because the IServiceCollection is provided, like in this:

https://github.com/Particular/docs.particular.net/commit/1caea8fb262655198e5d85fd4255ef61d95ef0af#diff-54f6c8671674829fafe0446884e7a16c

But outside that context it appears to be downright painful.

Reply:
in that case it looks good to me. i will do some research on the side and attempt to work out the underlying thinking behind the change